tests/components/ecobee/test_climate.py

```diff
@@ -1,288 +1,585 @@
+        "settings": {
+            "hvacMode": "cool",
+            "coolStages": 0,
+            "heatStages": 0,
+            "fanMinOnTime": 5,
+            "heatCoolMinDelta": 50,
+            "holdAction": "nextTransition",
+        },
+        "runtime": {
+            "connected": True,
+            "actualTemperature": 727,
+            "actualHumidity": 35,
+            "desiredHeat": 753,
+            "desiredCool": 734,
+            "desiredFanMode": "auto",
+        },
+        "events": [
+            {
+                "type": "sensor",
+                "name": "Living Room",
+                "running": True,
+                "holdClimateRef": "",
+            },
+            {
+                "type": "template",
+                "name": "_Default_",
+                "running": False,
+                "holdClimateRef": "",
+            },
+        ],
+        "program": {
+            "climates": [
+                {"name": "Home", "climateRef": "home"},
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "sleep",
+        },
+        "equipmentStatus": "auxHeat2,fan",
+    },
+]
+
+
+@pytest.fixture
+def mock_account(hass):
+    """Mock an ecobee account object."""
+    account = ecobee.EcobeeData(
+        hass,
+        MockConfigEntry(domain=ecobee.DOMAIN),
+        api_key="dummy",
+        refresh_token="dummy",
+    )
+
+    def mock_get_thermostat(index):
+        return account.ecobee.thermostats[index]
+
+    account.ecobee = Mock(
+        thermostats=MOCK_API_RESPONSE,
+        get_thermostat=mock_get_thermostat,
+        spec=pyecobee.Ecobee,
+    )
+
+    return account
+
+
+async def setup_climate(hass, mock_account):
+    """Load the ecobee climate platform with the mocked account."""
+    hass.config.components.add(ecobee.DOMAIN)
+    hass.data[ecobee.DOMAIN] = mock_account
+    config_entry = MockConfigEntry(domain=ecobee.DOMAIN)
+    await hass.config_entries.async_forward_entry_setup(config_entry, "climate")
+
+
+async def test_climate_platform_loads_correctly(hass, mock_account):
+    """Test that the climate platform loads with five climate entities."""
+    await setup_climate(hass, mock_account)
+    assert len(hass.states.async_all()) == 5
+    t_stat_1 = hass.states.get("climate.test_thermostat_1")
+    assert t_stat_1 is not None
+    t_stat_2 = hass.states.get("climate.test_thermostat_2")
+    assert t_stat_2 is not None
+    t_stat_3 = hass.states.get("climate.test_thermostat_3")
+    assert t_stat_3 is not None
+    t_stat_4 = hass.states.get("climate.test_thermostat_4")
+    assert t_stat_4 is not None
+    t_stat_5 = hass.states.get("climate.test_thermostat_5")
+    assert t_stat_5 is not None
+
+
+async def test_simple_properties(mock_account):
+    """Test that a climate entity returns expected values for simple properties."""
+    t_stat = Thermostat(mock_account, 0)
```

Comment:
Don't create the entity directly. Set up the component and platform and let the platform add an entity. Then assert the state of the entity via the core state machine. Patch and mock the library as needed to avoid I/O.

We should not interact directly with the entity in the tests. By using the core interfaces we make the tests robust for changes in the integration implementation.